### ISSUE TITLE
CyanogenMod 4.2.5 / 4.2.6 force-close

### ISSUE THREAD

**Issue by GoogleCodeExporter** [miramon98-1/android-sms#50]
```
What steps will reproduce the problem?
1. Open SMS Backup 
2. Select "backup now"
3.

What is the expected output? 
The program to begin backing up.

What do you see instead?
A force-close

What version of the product are you using? 
v1.1.0

On what operating system?
Android v2.0 (Motorola Droid/Shoals hardware)

Please provide any additional information below.
Sorry wish I had more info to give... let me know if I can provide anymore
info...

```

Original issue reported on code.google.com by `djxspike@gmail.com` on 10 Nov 2009 at 3:57

**Comment by GoogleCodeExporter** [comment]
```
same problem to me. the "processotv.studer.smssync" dosent' work! please make a 
new
update for this amazing software!
```

Original comment by `giovanni...@gmail.com` on 16 Nov 2009 at 10:30

**Comment by GoogleCodeExporter** [comment]
```
Same for me too :(

Using a HTC Magic with CyanogenMod 4.2.5 and kernel 2.6.29 with V1.6 of 
Androïd.

I had no problem with CyanogenMod 4.2.4, I think it's since it's creator 
included 
some v2.0 bits of code into it.

That's very sad that we have no more news from the dev side :(

```

Original comment by `d.paolino@gmail.com` on 17 Nov 2009 at 1:01

**Comment by GoogleCodeExporter** [comment]
```
If you can provide me with a log file, that would help a lot:

1. Download and install Log Collector from the Android Market:
http://www.cyrket.com/package/com.xtralogic.android.logcollector
2. Start SMS Backup and provoke the 'Force close'.
3. Start Log Collector and forward the log file to me. Please review
all the information in the log and remove any personal information
(passwords, phone numbers, names...) from it before sending it to me!

```

Original comment by `chstu...@gmail.com` on 17 Nov 2009 at 1:04
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
Original comment by `chstu...@gmail.com` on 17 Nov 2009 at 1:06

**Comment by GoogleCodeExporter** [comment]
```
From log collector:
11-17 20:54:19.967 D/SmsSync (16862): Sending 1 messages to server.
11-17 20:54:21.747 D/skia    (16647): purging 6K from font cache [1 entries]
11-17 20:54:21.867 D/dalvikvm(16862): GC freed 7991 objects / 471936 bytes in 
119ms
11-17 20:54:21.947 D/SmsSync (16862): Max synced date set to: 1258374513717
11-17 20:54:22.017 D/dalvikvm(16647): GC freed 1447 objects / 61552 bytes in 
274ms
11-17 20:54:22.027 E/DatabaseUtils(  340): Writing exception to parcel
11-17 20:54:22.027 E/DatabaseUtils(  340): java.lang.IllegalStateException: 
attempt to acquire a reference on a close 
SQLiteClosable
11-17 20:54:22.027 E/DatabaseUtils(  340): at 
android.database.sqlite.SQLiteClosable.acquireReference(SQLiteClosable.java:31)
11-17 20:54:22.027 E/DatabaseUtils(  340): at 
android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:69)
11-17 20:54:22.027 E/DatabaseUtils(  340): at 
android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:288)
11-17 20:54:22.027 E/DatabaseUtils(  340): at 
android.database.sqlite.SQLiteCursor.onMove(SQLiteCursor.java:260)
11-17 20:54:22.027 E/DatabaseUtils(  340): at 
android.database.AbstractCursor.moveToPosition(AbstractCursor.java:188)
11-17 20:54:22.027 E/DatabaseUtils(  340): at 
android.database.CursorToBulkCursorAdaptor.getWindow(CursorToBulkCursorAdaptor.j
ava:114)
11-17 20:54:22.027 E/DatabaseUtils(  340): at 
android.database.BulkCursorNative.onTransact(BulkCursorNative.java:66)
11-17 20:54:22.027 E/DatabaseUtils(  340): at 
android.os.Binder.execTransact(Binder.java:287)
11-17 20:54:22.027 E/DatabaseUtils(  340): at 
dalvik.system.NativeStart.run(Native Method)
11-17 20:54:22.067 I/ActivityManager(  152): Stopping service: 
tv.studer.smssync/.SmsSyncService
11-17 20:54:22.087 D/SmsSync (16862): Scheduled sync due in 1800 seconds.
11-17 20:54:22.107 W/dalvikvm(16862): threadid=15: thread exiting with uncaught 
exception (group=0x4001e170)
11-17 20:54:22.107 E/AndroidRuntime(16862): Uncaught handler: thread Thread-8 
exiting due to uncaught exception
11-17 20:54:22.127 E/AndroidRuntime(16862): java.lang.IllegalStateException: 
attempt to acquire a reference on a close 
SQLiteClosable
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
android.os.Parcel.readException(Parcel.java:1254)
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:145)
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:111)
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
android.database.BulkCursorProxy.getWindow(BulkCursorNative.java:217)
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
android.database.BulkCursorToCursorAdaptor.onMove(BulkCursorToCursorAdaptor.java
:91)
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
android.database.AbstractCursor.moveToPosition(AbstractCursor.java:188)
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
android.database.AbstractCursor.moveToNext(AbstractCursor.java:256)
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
android.database.CursorWrapper.moveToNext(CursorWrapper.java:191)
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
tv.studer.smssync.CursorToMessage.cursorToMessageArray(CursorToMessage.java:91)
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
tv.studer.smssync.SmsSyncService.backup(SmsSyncService.java:313)
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
tv.studer.smssync.SmsSyncService.access$2(SmsSyncService.java:239)
11-17 20:54:22.127 E/AndroidRuntime(16862): at 
tv.studer.smssync.SmsSyncService$1.run(SmsSyncService.java:156)
11-17 20:54:22.167 I/Process (  152): Sending signal. PID: 16862 SIG: 3
11-17 20:54:22.167 I/dalvikvm(16862): threadid=7: reacting to signal 3
11-17 20:54:22.197 I/dalvikvm(16862): Wrote stack trace to 
'/data/anr/traces.txt'
11-17 20:54:22.207 I/DumpStateReceiver(  152): Added state dump to 1 crashes

Contents of the crash in /data/anr/traces.txt:
----- pid 16862 at 2009-11-17 20:54:22 -----
Cmd line: tv.studer.smssync

DALVIK THREADS:
"main" prio=5 tid=3 WAIT
  | group="main" sCount=1 dsCount=0 s=N obj=0x4001e250 self=0xbc20
  | sysTid=16862 nice=0 sched=0/0 cgrp=default handle=-1344001384
  at java.lang.Object.wait(Native Method)
  - waiting on <0x1cd3f8> (a android.os.MessageQueue)
  at java.lang.Object.wait(Object.java:288)
  at android.os.MessageQueue.next(MessageQueue.java:148)
  at android.os.Looper.loop(Looper.java:110)
  at android.app.ActivityThread.main(ActivityThread.java:4203)
  at java.lang.reflect.Method.invokeNative(Native Method)
  at java.lang.reflect.Method.invoke(Method.java:521)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:791)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:549)
  at dalvik.system.NativeStart.main(Native Method)

"Thread-8" prio=5 tid=15 NATIVE
  | group="main" sCount=1 dsCount=0 s=N obj=0x43828a98 self=0x1ecec8
  | sysTid=16868 nice=10 sched=0/0 cgrp=bg_non_interactive handle=2019296
  at android.os.BinderProxy.transact(Native Method)
  at android.app.ActivityManagerProxy.handleApplicationError(ActivityManagerNative.java:2243)
  at com.android.internal.os.RuntimeInit.crash(RuntimeInit.java:302)
  at com.android.internal.os.RuntimeInit$UncaughtHandler.uncaughtException(RuntimeInit.java:75)
  at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:887)
  at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:884)

"Binder Thread #2" prio=5 tid=13 NATIVE
  | group="main" sCount=1 dsCount=0 s=N obj=0x437edf18 self=0x175fe0
  | sysTid=16867 nice=0 sched=0/0 cgrp=default handle=1531808
  at dalvik.system.NativeStart.run(Native Method)

"Binder Thread #1" prio=5 tid=11 NATIVE
  | group="main" sCount=1 dsCount=0 s=N obj=0x437ede58 self=0x1752d8
  | sysTid=16866 nice=0 sched=0/0 cgrp=default handle=1182088
  at dalvik.system.NativeStart.run(Native Method)

"JDWP" daemon prio=5 tid=9 VMWAIT
  | group="system" sCount=1 dsCount=0 s=N obj=0x437e82a0 self=0x122448
  | sysTid=16865 nice=0 sched=0/0 cgrp=default handle=1188872
  at dalvik.system.NativeStart.run(Native Method)

"Signal Catcher" daemon prio=5 tid=7 RUNNABLE
  | group="system" sCount=0 dsCount=0 s=N obj=0x437e81e8 self=0x122158
  | sysTid=16864 nice=0 sched=0/0 cgrp=default handle=1188120
  at dalvik.system.NativeStart.run(Native Method)

"HeapWorker" daemon prio=5 tid=5 VMWAIT
  | group="system" sCount=1 dsCount=0 s=N obj=0x4289d128 self=0x121710
  | sysTid=16863 nice=0 sched=0/0 cgrp=default handle=1185488
  at dalvik.system.NativeStart.run(Native Method)

----- end 16862 -----
```

Original comment by `balloob` on 17 Nov 2009 at 8:00

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I hope can be useful:
11-17 12:44:07.225 D/NetworkLocationProvider(  140): onCellLocationChanged 
[30031,8784]
11-17 12:44:07.235 I/ActivityManager(  140): Start proc com.android.mms for 
broadcast
com.android.mms/.transaction.MmsSystemEventReceiver: pid=1192 uid=10024 
gids={3003, 1015}
11-17 12:44:07.555 I/ActivityManager(  140): Process android.process.media (pid 
985)
has died.
11-17 12:44:07.865 D/dalvikvm( 1121): GC freed 11375 objects / 481952 bytes in 
96ms
11-17 12:44:17.938 E/SmsBackup-Backend( 1121): Failed to login, closing 
connection
11-17 12:44:17.985 I/SmsSync ( 1121):
11-17 12:44:17.985 I/SmsSync ( 1121):
tv.studer.smssync.SmsSyncService$AuthenticationErrorException: IO Error
11-17 12:44:17.985 I/SmsSync ( 1121): at
tv.studer.smssync.SmsSyncService.backup(SmsSyncService.java:300)
11-17 12:44:17.985 I/SmsSync ( 1121): at
tv.studer.smssync.SmsSyncService.access$2(SmsSyncService.java:239)
11-17 12:44:17.985 I/SmsSync ( 1121): at
tv.studer.smssync.SmsSyncService$1.run(SmsSyncService.java:156)
11-17 12:44:17.985 I/SmsSync ( 1121): Caused by:
com.android.email.mail.MessagingException: IO Error
11-17 12:44:17.985 I/SmsSync ( 1121): at
com.android.email.mail.store.ImapStore$ImapFolder.ioExceptionHandler(ImapStore.j
ava:1264)
11-17 12:44:17.985 I/SmsSync ( 1121): at
com.android.email.mail.store.ImapStore$ImapFolder.exists(ImapStore.java:485)
11-17 12:44:17.985 I/SmsSync ( 1121): at
tv.studer.smssync.SmsSyncService.backup(SmsSyncService.java:293)
11-17 12:44:17.985 I/SmsSync ( 1121): ... 2 more
11-17 12:44:17.985 I/SmsSync ( 1121): Caused by: 
java.net.SocketTimeoutException:
Socket is not connected
11-17 12:44:17.985 I/SmsSync ( 1121): at
org.apache.harmony.luni.platform.OSNetworkSystem.connectStreamWithTimeoutSocketI
mpl(Native
Method)
11-17 12:44:17.985 I/SmsSync ( 1121): at
org.apache.harmony.luni.platform.OSNetworkSystem.connectStreamWithTimeoutSocket(
OSNetworkSystem.java:131)
11-17 12:44:17.985 I/SmsSync ( 1121): at
org.apache.harmony.luni.net.PlainSocketImpl.connect(PlainSocketImpl.java:247)
11-17 12:44:17.985 I/SmsSync ( 1121): at
org.apache.harmony.luni.net.PlainSocketImpl.connect(PlainSocketImpl.java:535)
11-17 12:44:17.985 I/SmsSync ( 1121): at 
java.net.Socket.connect(Socket.java:1054)
11-17 12:44:17.985 I/SmsSync ( 1121): at
com.android.email.mail.store.ImapStore$ImapConnection.open(ImapStore.java:1322)
11-17 12:44:17.985 I/SmsSync ( 1121): at
com.android.email.mail.store.ImapStore$ImapConnection.sendCommand(ImapStore.java
:1467)
11-17 12:44:17.985 I/SmsSync ( 1121): at
com.android.email.mail.store.ImapStore$ImapConnection.executeSimpleCommand(ImapS
tore.java:1514)
11-17 12:44:17.985 I/SmsSync ( 1121): at
com.android.email.mail.store.ImapStore$ImapConnection.executeSimpleCommand(ImapS
tore.java:1505)
11-17 12:44:17.985 I/SmsSync ( 1121): at
com.android.email.mail.store.ImapStore$ImapFolder.exists(ImapStore.java:476)
11-17 12:44:17.985 I/SmsSync ( 1121): ... 3 more
11-17 12:44:18.025 I/ActivityManager(  140): Stopping service:
tv.studer.smssync/.SmsSyncService
11-17 12:44:18.045 D/SmsSync ( 1121): Not scheduling sync because auto sync is 
disabled.
11-17 12:44:19.885 I/SmsSync ( 1121): Starting backup...
11-17 12:44:19.975 D/SmsSync ( 1121): Total messages to backup: 100
11-17 12:44:19.985 I/SmsBackup-Backend( 1121): Connecting to imap.gmail.com @ 
IP addr
imap.gmail.com/74.125.43.109:993
11-17 12:44:21.045 I/global  ( 1121): Default buffer size used in
BufferedOutputStream constructor. It would be better to be explicit if an 8k 
buffer
is required.
11-17 12:44:22.115 D/dalvikvm( 1156): GC freed 7573 objects / 413192 bytes in 
101ms
11-17 12:44:22.375 D/dalvikvm( 1121): GC freed 8259 objects / 445864 bytes in 
127ms
11-17 12:44:22.695 D/SmsSync ( 1121): Sending 1 messages to server.
11-17 12:44:23.685 D/dalvikvm( 1156): GC freed 8960 objects / 532856 bytes in 
99ms
11-17 12:44:25.595 D/SmsSync ( 1121): Max synced date set to: 1258300394017
11-17 12:44:25.645 E/DatabaseUtils(  180): Writing exception to parcel
11-17 12:44:25.645 E/DatabaseUtils(  180): java.lang.IllegalStateException: 
attempt
to acquire a reference on a close SQLiteClosable
11-17 12:44:25.645 E/DatabaseUtils(  180): at
android.database.sqlite.SQLiteClosable.acquireReference(SQLiteClosable.java:31)
11-17 12:44:25.645 E/DatabaseUtils(  180): at
android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:69)
11-17 12:44:25.645 E/DatabaseUtils(  180): at
android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:288)
11-17 12:44:25.645 E/DatabaseUtils(  180): at
android.database.sqlite.SQLiteCursor.onMove(SQLiteCursor.java:260)
11-17 12:44:25.645 E/DatabaseUtils(  180): at
android.database.AbstractCursor.moveToPosition(AbstractCursor.java:188)
11-17 12:44:25.645 E/DatabaseUtils(  180): at
android.database.CursorToBulkCursorAdaptor.getWindow(CursorToBulkCursorAdaptor.j
ava:114)
11-17 12:44:25.645 E/DatabaseUtils(  180): at
android.database.BulkCursorNative.onTransact(BulkCursorNative.java:66)
11-17 12:44:25.645 E/DatabaseUtils(  180): at
android.os.Binder.execTransact(Binder.java:287)
11-17 12:44:25.645 E/DatabaseUtils(  180): at 
dalvik.system.NativeStart.run(Native
Method)
11-17 12:44:25.665 I/ActivityManager(  140): Stopping service:
tv.studer.smssync/.SmsSyncService
11-17 12:44:25.675 D/SmsSync ( 1121): Not scheduling sync because auto sync is 
disabled.
11-17 12:44:25.695 W/dalvikvm( 1121): threadid=15: thread exiting with uncaught
exception (group=0x4001e170)
11-17 12:44:25.695 E/AndroidRuntime( 1121): Uncaught handler: thread Thread-9 
exiting
due to uncaught exception
11-17 12:44:25.705 E/AndroidRuntime( 1121): java.lang.IllegalStateException: 
attempt
to acquire a reference on a close SQLiteClosable
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
android.os.Parcel.readException(Parcel.java:1254)
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:145)
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:111)
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
android.database.BulkCursorProxy.getWindow(BulkCursorNative.java:217)
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
android.database.BulkCursorToCursorAdaptor.onMove(BulkCursorToCursorAdaptor.java
:91)
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
android.database.AbstractCursor.moveToPosition(AbstractCursor.java:188)
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
android.database.AbstractCursor.moveToNext(AbstractCursor.java:256)
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
android.database.CursorWrapper.moveToNext(CursorWrapper.java:191)
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
tv.studer.smssync.CursorToMessage.cursorToMessageArray(CursorToMessage.java:91)
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
tv.studer.smssync.SmsSyncService.backup(SmsSyncService.java:313)
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
tv.studer.smssync.SmsSyncService.access$2(SmsSyncService.java:239)
11-17 12:44:25.705 E/AndroidRuntime( 1121): at
tv.studer.smssync.SmsSyncService$1.run(SmsSyncService.java:156)
11-17 12:44:25.725 I/Process (  140): Sending signal. PID: 1121 SIG: 3
11-17 12:44:25.725 I/dalvikvm( 1121): threadid=7: reacting to signal 3
11-17 12:44:25.765 I/dalvikvm( 1121): Wrote stack trace to 
'/data/anr/traces.txt'
11-17 12:44:25.965 I/DumpStateReceiver(  140): Added state dump to 1 crashes
11-17 12:44:26.215 I/ActivityManager(  140): Process com.android.mms (pid 1192) 
has died.
11-17 12:44:28.075 I/Process ( 1121): Sending signal. PID: 1121 SIG: 9
11-17 12:44:28.095 I/ActivityManager(  140): Process tv.studer.smssync (pid 
1121) has
died.
11-17 12:44:28.105 I/WindowManager(  140): WIN DEATH: Window{439603f0
tv.studer.smssync/tv.studer.smssync.SmsSync paused=false}
11-17 12:44:28.135 W/UsageStats(  140): Unexpected resume of 
com.android.launcher
while already resumed in tv.studer.smssync
11-17 12:44:28.255 W/InputManagerService(  140): Got RemoteException sending
setActive(false) notification to pid 1121 uid 10041
11-17 12:44:30.865 D/dalvikvm(  183): GC freed 17759 objects / 967464 bytes in 
196ms
11-17 12:44:36.375 D/dalvikvm(  140): threadid=15: bogus mon 1+0>0; adjusting
11-17 12:44:38.885 I/ActivityManager(  140): Starting activity: Intent {
act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] 
flg=0x10200000
cmp=com.xtralogic.android.logcollector/.SendLogActivity }
11-17 12:44:38.955 I/ActivityManager(  140): Start proc
com.xtralogic.android.logcollector for activity
com.xtralogic.android.logcollector/.SendLogActivity: pid=1211 uid=10061 
gids={1007, 1015}
11-17 12:44:39.175 D/dalvikvm(  123): GC freed 303 objects / 11576 bytes in 
220ms
11-17 12:44:39.395 D/dalvikvm(  123): GC freed 57 objects / 2568 bytes in 179ms
11-17 12:44:39.535 D/dalvikvm(  123): GC freed 2 objects / 56 bytes in 137ms
11-17 12:44:41.325 D/dalvikvm(  140): GC freed 13858 objects / 1191912 bytes in 
200ms
11-17 12:44:41.555 D/OpenSSLSessionImpl(  140): Freeing OpenSSL session
11-17 12:44:41.645 I/global  ( 1211): Default buffer size used in BufferedReader
constructor. It would be better to be explicit if an 8k-char buffer is required.
11-17 12:44:41.685 I/ActivityManager(  140): Displayed activity
com.xtralogic.android.logcollector/.SendLogActivity: 2744 ms (total 2744 ms)
11-17 12:44:41.985 D/dalvikvm( 1211): GC freed 1589 objects / 234936 bytes in
125ms[/spoiler]
```

Original comment by `giovanni...@gmail.com` on 18 Nov 2009 at 2:28

**Comment by GoogleCodeExporter** [comment]
```
Thank you, I'll have a look as soon as possible!
```

Original comment by `chstu...@gmail.com` on 19 Nov 2009 at 8:51

**Comment by GoogleCodeExporter** [comment]
```
Issue 51 has been merged into this issue.
```

Original comment by `chstu...@gmail.com` on 19 Nov 2009 at 9:04

**Comment by GoogleCodeExporter** [comment]
```
Also experiencing this on Cyanogen 4.2.5. Please let me know if there is 
anything
else needed to solve this. I'd be glad to provide debug logs and test possible
solutions if that's helpful.

Thanks for a useful program.
```

Original comment by `alerque` on 20 Nov 2009 at 8:14

**Comment by GoogleCodeExporter** [comment]
```
My logs are from Cyanogen 4.2.5 btw.
```

Original comment by `balloob` on 20 Nov 2009 at 9:39

**Comment by GoogleCodeExporter** [comment]
```
same problem here... also running cyanogenmod 4.2.5
```

Original comment by `mat.fr...@googlemail.com` on 21 Nov 2009 at 8:04

**Comment by GoogleCodeExporter** [comment]
```
Also getting this on Cyanogen 4.2.5, here's a log:

I/SmsSync ( 1862): Starting backup...
D/SmsSync ( 1862): Total messages to backup: 2
I/SmsBackup-Backend( 1862): Connecting to imap.gmail.com @ IP addr 
imap.gmail.com/74.125.77.109:993
V/SmsSync ( 1862): Looked up unknown address: +4542423141
D/SmsSync ( 1862): Sending 1 messages to server.
D/SmsSync ( 1862): Max synced date set to: 1258901203476
E/DatabaseUtils(  191): Writing exception to parcel
E/DatabaseUtils(  191): java.lang.IllegalStateException: attempt to acquire a 
reference on a close SQLiteClosable
E/DatabaseUtils(  191):     at 
android.database.sqlite.SQLiteClosable.acquireReference(SQLiteClosable.java:31)
E/DatabaseUtils(  191):     at android.database.sqlite.SQLiteQuery.fillWindow
(SQLiteQuery.java:69)
E/DatabaseUtils(  191):     at android.database.sqlite.SQLiteCursor.fillWindow
(SQLiteCursor.java:288)
E/DatabaseUtils(  191):     at android.database.sqlite.SQLiteCursor.onMove
(SQLiteCursor.java:260)
E/DatabaseUtils(  191):     at android.database.AbstractCursor.moveToPosition
(AbstractCursor.java:188)
E/DatabaseUtils(  191):     at 
android.database.CursorToBulkCursorAdaptor.getWindow
(CursorToBulkCursorAdaptor.java:114)
E/DatabaseUtils(  191):     at android.database.BulkCursorNative.onTransact
(BulkCursorNative.java:66)
E/DatabaseUtils(  191):     at android.os.Binder.execTransact(Binder.java:287)
E/DatabaseUtils(  191):     at dalvik.system.NativeStart.run(Native Method)
I/ActivityManager(  145): Stopping service: tv.studer.smssync/.SmsSyncService
D/SmsSync ( 1862): Scheduled sync due in 1800 seconds.
W/dalvikvm( 1862): threadid=15: thread exiting with uncaught exception 
(group=0x4001e170)
E/AndroidRuntime( 1862): Uncaught handler: thread Thread-8 exiting due to 
uncaught 
exception
I/DumpStateReceiver(  145): Added state dump to 1 crashes
E/AndroidRuntime( 1862): java.lang.IllegalStateException: attempt to acquire a 
reference on a close SQLiteClosable
E/AndroidRuntime( 1862):    at android.os.Parcel.readException(Parcel.java:1254)
E/AndroidRuntime( 1862):    at 
android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:145)
E/AndroidRuntime( 1862):    at 
android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:111)
E/AndroidRuntime( 1862):    at android.database.BulkCursorProxy.getWindow
(BulkCursorNative.java:217)
E/AndroidRuntime( 1862):    at android.database.BulkCursorToCursorAdaptor.onMove
(BulkCursorToCursorAdaptor.java:91)
E/AndroidRuntime( 1862):    at android.database.AbstractCursor.moveToPosition
(AbstractCursor.java:188)
E/AndroidRuntime( 1862):    at android.database.AbstractCursor.moveToNext
(AbstractCursor.java:256)
E/AndroidRuntime( 1862):    at android.database.CursorWrapper.moveToNext
(CursorWrapper.java:191)
E/AndroidRuntime( 1862):    at 
tv.studer.smssync.CursorToMessage.cursorToMessageArray(CursorToMessage.java:91)
E/AndroidRuntime( 1862):    at tv.studer.smssync.SmsSyncService.backup
(SmsSyncService.java:313)
E/AndroidRuntime( 1862):    at tv.studer.smssync.SmsSyncService.access
$2(SmsSyncService.java:239)
E/AndroidRuntime( 1862):    at tv.studer.smssync.SmsSyncService$1.run
(SmsSyncService.java:156)
I/Process (  145): Sending signal. PID: 1862 SIG: 3
I/dalvikvm( 1862): threadid=7: reacting to signal 3
I/dalvikvm( 1862): Wrote stack trace to '/data/anr/traces.txt'
D/skia    ( 1780): purging 166K from font cache [22 entries]
I/dalvikvm( 1780): Uncaught exception thrown by finalizer (will be discarded):
I/dalvikvm( 1780): Ljava/lang/IllegalStateException;: Finalizing cursor 
android.database.sqlite.SQLiteCursor@4387a5e8 on null that has not been 
deactivated 
or closed
I/dalvikvm( 1780):  at android.database.sqlite.SQLiteCursor.finalize
(SQLiteCursor.java:596)
I/dalvikvm( 1780):  at dalvik.system.NativeStart.run(Native Method)
I/Process ( 1862): Sending signal. PID: 1862 SIG: 9
I/ActivityManager(  145): Process tv.studer.smssync (pid 1862) has died.
I/WindowManager(  145): WIN DEATH: Window{43a48618 tv.studer.smssync/
tv.studer.smssync.SmsSync paused=false}
W/UsageStats(  145): Unexpected resume of com.irrenhaus.advancedlauncher.donut 
while 
already resumed in tv.studer.smssync
I/ActivityManager(  145): Process android.process.acore (pid 1873) has died.
I/ActivityManager(  145): Low Memory: No more background processes.
W/InputManagerService(  145): Got RemoteException sending setActive(false) 
notification to pid 1862 uid 10077
```

Original comment by `martin.e...@gmail.com` on 22 Nov 2009 at 2:52

**Comment by GoogleCodeExporter** [comment]
```
Issue 54 has been merged into this issue.
```

Original comment by `chstu...@gmail.com` on 23 Nov 2009 at 11:53

**Comment by GoogleCodeExporter** [comment]
```
Issue 53 has been merged into this issue.
```

Original comment by `chstu...@gmail.com` on 23 Nov 2009 at 3:12

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I was unable to reproduce this on a stock 2.0 image. Anybody on a Droid or 
another 2.0 
phone who can confirm this is not a problem only with Cyanogen?
```

Original comment by `chstu...@gmail.com` on 26 Nov 2009 at 6:45

**Comment by GoogleCodeExporter** [comment]
```
As I could not find any instructions on how to let CyanogenMod 4.2.5 run in an
emulator, I compiled a patched version of SMS Backup to hopefully include a bit 
more
information as to why this crash is happening.

Could somebody with Cyanogen 4.2.5 please give it a try?

It might still not be enough, in which case I'll have to flash my G1 or compile 
a new
version.
```

Original comment by `chstu...@gmail.com` on 26 Nov 2009 at 8:52

Attachments:
- [SmsSync-CyanogenTesting01.apk](https://storage.googleapis.com/google-code-attachments/android-sms/issue-50/comment-18/SmsSync-CyanogenTesting01.apk)
- [cyanogen-testing01.patch](https://storage.googleapis.com/google-code-attachments/android-sms/issue-50/comment-18/cyanogen-testing01.patch)

**Comment by GoogleCodeExporter** [comment]
```
I just tried to install that APK onto my cm 4.2.5 ADP1. From the phone it says 
"this packages cannot be installed". Trying to install 
from adb shows: Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES]
```

Original comment by `alerque` on 26 Nov 2009 at 9:02

**Comment by GoogleCodeExporter** [comment]
```
Sorry about that. Hope this attachment is working now!

```

Original comment by `chstu...@gmail.com` on 27 Nov 2009 at 11:33

Attachments:
- [SmsSync-CyanogenTesting01-Signed.apk](https://storage.googleapis.com/google-code-attachments/android-sms/issue-50/comment-20/SmsSync-CyanogenTesting01-Signed.apk)

**Comment by GoogleCodeExporter** [comment]
```
i just installed the new version, now the installation works (you have to 
uninstall 
the old version first!, if you du not you get an
[INSTALL_PARSE_FAILED_INCONSISTENT_CERTIFICATES] via adb install)

but the error message stays the same, how can i send you the additional 
information 
which this version should provide?
```

Original comment by `mat.fr...@googlemail.com` on 27 Nov 2009 at 12:43

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
The stack trace reported in the log file should be different now. Please catch 
it
using the 
log collector and attach the relevant bits to this bug.

Thanks!
```

Original comment by `chstu...@gmail.com` on 27 Nov 2009 at 2:10

**Comment by GoogleCodeExporter** [comment]
```
don't know whats relevant so i simply attach the log at the time when sms 
backup 
crashed (+/-30 sec)
11-27 15:48:39.311 I/ActivityManager(   97): Starting activity: Intent 
{ act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] 
flg=0x10200000 cmp=tv.studer.smssync/.SmsSync }
11-27 15:48:39.351 W/UsageStats(   97): Something wrong here, didn't expect 
com.android.launcher to be paused
11-27 15:48:39.491 I/ActivityManager(   97): Start proc tv.studer.smssync for 
activity tv.studer.smssync/.SmsSync: pid=6718 uid=10060 gids={3003, 1015}
11-27 15:48:39.591 D/dalvikvm(   80): GC freed 298 objects / 11344 bytes in 
190ms
11-27 15:48:39.841 D/dalvikvm(   80): GC freed 57 objects / 2528 bytes in 233ms
11-27 15:48:40.011 D/dalvikvm(   80): GC freed 2 objects / 56 bytes in 168ms
11-27 15:48:40.191 I/ActivityManager(   97): Starting activity: Intent 
{ act=android.intent.action.VIEW dat=content://sms/2237 flg=0x10000000 
cmp=com.p1.chompsms/.activities.QuickReply (has extras) }
11-27 15:48:40.561 I/ActivityManager(   97): Stopping service: 
com.android.mms/.transaction.SmsReceiverService
11-27 15:48:40.681 I/ActivityManager(   97): Process com.android.mms (pid 6702) 
has 
died.
11-27 15:48:40.691 I/ActivityManager(   97): Low Memory: No more background 
processes.
11-27 15:48:40.701 W/ActivityManager(   97): Activity pause timeout for 
HistoryRecord
{439bd970 tv.studer.smssync/.SmsSync}
11-27 15:48:41.161 I/ActivityManager(   97): Start proc android.process.media 
for 
content provider com.android.providers.media/.MediaProvider: pid=6727 uid=10021 
gids=
{1015, 2001, 3003, 1006}
11-27 15:48:41.601 I/ActivityThread( 6727): Publishing provider media: 
com.android.providers.media.MediaProvider
11-27 15:48:41.641 D/dalvikvm( 6718): GC freed 780 objects / 57360 bytes in 
351ms
11-27 15:48:41.941 V/MediaProvider( 6727): Attached volume: internal
11-27 15:48:41.991 V/MediaProvider( 6727): /sdcard volume ID: 14992304
11-27 15:48:42.021 W/Resources( 6718): Converting to string: TypedValue{t=0x10/
d=0x64 a=-1}
11-27 15:48:42.321 I/ActivityManager(   97): Process com.weather.Weather (pid 
5223) 
has died.
11-27 15:48:42.321 W/ActivityManager(   97): Scheduling restart of crashed 
service 
com.weather.Weather/com.weather.service.WeatherService in 15596ms
11-27 15:48:42.351 I/ActivityManager(   97): Low Memory: No more background 
processes.
11-27 15:48:45.371 I/Bluetooth AT sent(  142): +CIEV: 5,3
11-27 15:48:45.401 V/MediaProvider( 6727): Attached volume: external
11-27 15:48:45.701 I/ActivityThread( 6727): Publishing provider downloads: 
com.android.providers.downloads.DownloadProvider
11-27 15:48:45.861 I/ActivityThread( 6727): Publishing provider drm: 
com.android.providers.drm.DrmProvider
11-27 15:48:45.861 I/ActivityManager(   97): Displayed activity 
com.p1.chompsms/.activities.QuickReply: 5161 ms (total 6484 ms)
11-27 15:48:46.081 I/ActivityManager(   97): Displayed activity 
tv.studer.smssync/.SmsSync: 6700 ms (total 6700 ms)
11-27 15:48:46.461 D/dalvikvm(  142): GC freed 3179 objects / 147800 bytes in 
190ms
11-27 15:48:47.951 W/AudioFlinger(   81): write blocked for 105 msecs
11-27 15:48:48.411 I/ActivityManager(   97): Start proc 
com.google.process.gapps for 
service com.google.android.apps.gtalkservice/
com.google.android.gtalkservice.service.GTalkService: pid=6739 uid=10004 
gids={3003}
11-27 15:48:48.421 I/ActivityManager(   97): Process android.process.media (pid 
6727) has died.
11-27 15:48:48.421 I/ActivityManager(   97): Low Memory: No more background 
processes.
11-27 15:48:48.631 I/ActivityThread( 6739): Publishing provider 
com.google.settings: 
com.google.android.providers.settings.GoogleSettingsProvider
11-27 15:48:48.711 I/ActivityThread( 6739): Publishing provider gmail-ls: 
com.google.android.providers.gmail.MailProvider
11-27 15:48:48.731 I/ActivityManager(   97): Stopping service: 
com.p1.chompsms/.sms.SmsReceiverService
11-27 15:48:48.891 I/SmsSync ( 6718): Starting backup...
11-27 15:48:48.971 I/WindowManager(   97): WIN DEATH: Window{439e2cc0 
com.p1.chompsms/com.p1.chompsms.activities.ConversationList paused=false}
11-27 15:48:48.971 I/ActivityManager(   97): Process com.p1.chompsms (pid 6650) 
has 
died.
11-27 15:48:48.971 I/ActivityManager(   97): Low Memory: No more background 
processes.
11-27 15:48:49.471 D/SmsSync ( 6718): Total messages to backup: 2
11-27 15:48:51.031 I/SmsBackup-Backend( 6718): Connecting to imap.gmail.com @ 
IP 
addr imap.gmail.com/209.85.135.109:993
11-27 15:48:51.861 D/GTalkService( 6739): [GTalkImSession] login: 
mat.frank@googlemail.com, hostConn.connState = RECONNECTION_SCHEDULED, 
hostConn.error=NO ERROR
11-27 15:48:52.591 D/dalvikvm( 6739): GC freed 4016 objects / 310232 bytes in 
106ms
11-27 15:48:52.961 D/gmail-ls( 6739): Purging old attachments
11-27 15:48:53.111 D/dalvikvm( 6718): GC freed 5384 objects / 263224 bytes in 
98ms
11-27 15:48:53.381 I/gmail-ls( 6739): 
calculateUnknownSyncRationalesAndPurgeInBackground: queueing
11-27 15:48:53.381 I/gmail-ls( 6739): 
calculateUnknownSyncRationalesAndPurgeInBackground: running
11-27 15:48:54.471 D/dalvikvm(   97): GC freed 6780 objects / 304728 bytes in 
333ms
11-27 15:48:54.921 D/skia    (  145): purging 187K from font cache [25 entries]
11-27 15:48:54.961 D/GTalkService( 6739): [RosterMgr] roster not modified
11-27 15:48:55.181 D/dalvikvm( 6718): GC freed 13835 objects / 527632 bytes in 
189ms
11-27 15:48:55.261 D/dalvikvm(  145): GC freed 1554 objects / 85992 bytes in 
334ms
11-27 15:48:55.451 D/GTalkService( 6739): [OtrMgr] OTR not modified
11-27 15:48:55.861 D/dalvikvm( 6718): GC freed 10507 objects / 443096 bytes in 
93ms
11-27 15:48:56.241 D/SmsSync ( 6718): SmsSyncService.onStart(): Already running.
11-27 15:48:56.931 D/dalvikvm( 6718): GC freed 9333 objects / 473936 bytes in 
97ms
11-27 15:48:57.571 D/dalvikvm( 6718): GC freed 8924 objects / 522928 bytes in 
101ms
11-27 15:48:57.951 I/ActivityManager(   97): Start proc com.weather.Weather for 
service com.weather.Weather/com.weather.service.WeatherService: pid=6764 
uid=10064 
gids={3003, 1015}
11-27 15:48:58.621 I/ActivityThread( 6764): Publishing provider 
com.weather.provider.twcWeatherDataPoints: 
com.weather.Weather.twcWeatherDataPointsProvider
11-27 15:48:58.711 D/dalvikvm( 6718): GC freed 9914 objects / 564184 bytes in 
168ms
11-27 15:48:58.991 I/ActivityThread( 6764): Publishing provider 
com.weather.provider.twcWeatherData: com.weather.Weather.twcWeatherDataProvider
11-27 15:48:59.051 D/WeatherService( 6764): goActive
11-27 15:48:59.071 D/WeatherService( 6764): getWeatherUpdate
11-27 15:48:59.841 D/WeatherGadgetProvider80x100( 6764): updateGadget 
gadgetId=39
11-27 15:49:00.091 D/WeatherGadgetProvider80x100( 6764): updateGadget 
gadgetId=40
11-27 15:49:00.121 D/WeatherService( 6764): stopTracking
11-27 15:49:00.481 D/dalvikvm( 6764): GC freed 3720 objects / 216752 bytes in 
111ms
11-27 15:49:00.851 D/WeatherGadgetProvider160x200( 6764): updateGadget 
gadgetId=34
11-27 15:49:01.151 D/dalvikvm( 6764): GC freed 7366 objects / 444536 bytes in 
127ms
11-27 15:49:01.391 I/global  ( 6718): Default buffer size used in 
BufferedOutputStream constructor. It would be better to be explicit if an 8k 
buffer 
is required.
11-27 15:49:02.431 D/dalvikvm( 6718): GC freed 8406 objects / 474504 bytes in 
127ms
11-27 15:49:02.811 D/SmsSync ( 6718): Sending 1 messages to server.
11-27 15:49:06.191 D/SmsSync ( 6718): Max synced date set to: 1259333303950
11-27 15:49:06.191 I/ActivityManager(   97): Stopping service: 
tv.studer.smssync/.SmsSyncService
11-27 15:49:06.221 D/SmsSync ( 6718): Scheduled sync due in 1800 seconds.
11-27 15:49:06.261 W/dalvikvm( 6718): threadid=17: thread exiting with uncaught 
exception (group=0x4001e170)
11-27 15:49:06.261 E/AndroidRuntime( 6718): Uncaught handler: thread Thread-9 
exiting due to uncaught exception
11-27 15:49:06.301 E/AndroidRuntime( 6718): java.lang.RuntimeException: 
Invalidated. 
Boom!
11-27 15:49:06.301 E/AndroidRuntime( 6718): at tv.studer.smssync.SmsSyncService
$2.onInvalidated(SmsSyncService.java:272)
11-27 15:49:06.301 E/AndroidRuntime( 6718): at 
android.database.DataSetObservable.notifyInvalidated(DataSetObservable.java:43)
11-27 15:49:06.301 E/AndroidRuntime( 6718): at 
android.database.AbstractCursor.deactivateInternal(AbstractCursor.java:89)
11-27 15:49:06.301 E/AndroidRuntime( 6718): at 
android.database.AbstractCursor.close
(AbstractCursor.java:108)
11-27 15:49:06.301 E/AndroidRuntime( 6718): at 
android.database.BulkCursorToCursorAdaptor.close(BulkCursorToCursorAdaptor.java:
123)
11-27 15:49:06.301 E/AndroidRuntime( 6718): at 
android.database.CursorWrapper.close
(CursorWrapper.java:45)
11-27 15:49:06.301 E/AndroidRuntime( 6718): at android.content.ContentResolver
$CursorWrapperInner.close(ContentResolver.java:746)
11-27 15:49:06.301 E/AndroidRuntime( 6718): at 
tv.studer.smssync.SmsSyncService.backup(SmsSyncService.java:351)
11-27 15:49:06.301 E/AndroidRuntime( 6718): at 
tv.studer.smssync.SmsSyncService.access$2(SmsSyncService.java:240)
11-27 15:49:06.301 E/AndroidRuntime( 6718): at 
tv.studer.smssync.SmsSyncService$1.run
(SmsSyncService.java:157)
11-27 15:49:06.341 I/Process (   97): Sending signal. PID: 6718 SIG: 3
11-27 15:49:06.341 I/dalvikvm( 6718): threadid=7: reacting to signal 3
11-27 15:49:06.581 I/dalvikvm( 6718): Wrote stack trace to 
'/data/anr/traces.txt'
11-27 15:49:08.671 I/Process ( 6718): Sending signal. PID: 6718 SIG: 9
11-27 15:49:08.681 I/ActivityManager(   97): Process tv.studer.smssync (pid 
6718) 
has died.
11-27 15:49:08.711 W/UsageStats(   97): Unexpected resume of 
com.android.launcher 
while already resumed in tv.studer.smssync
11-27 15:49:08.741 I/WindowManager(   97): WIN DEATH: Window{4372c158 
tv.studer.smssync/tv.studer.smssync.SmsSync paused=false}
11-27 15:49:08.751 I/ActivityManager(   97): Low Memory: No more background 
processes.
11-27 15:49:08.941 W/InputManagerService(   97): Got RemoteException sending 
setActive(false) notification to pid 6718 uid 10060
11-27 15:49:11.011 D/dalvikvm(  145): GC freed 7008 objects / 412344 bytes in 
133ms
11-27 15:49:11.251 D/dalvikvm(  145): GC freed 114 objects / 6456 bytes in 123ms
11-27 15:49:12.881 D/dalvikvm(  145): GC freed 33 objects / 1480 bytes in 139ms
11-27 15:49:14.301 D/skia    (   97): purging 167K from font cache [20 entries]
11-27 15:49:14.641 D/dalvikvm(   97): GC freed 3545 objects / 162328 bytes in 
340ms
11-27 15:49:15.831 I/ActivityManager(   97): Starting activity: Intent 
{ act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] 
flg=0x10200000 cmp=com.xtralogic.android.logcollector/.SendLogActivity }
11-27 15:49:15.931 I/ActivityManager(   97): Start proc 
com.xtralogic.android.logcollector for activity 
com.xtralogic.android.logcollector/.SendLogActivity: pid=6783 uid=10075 
gids={1007, 
1015}
11-27 15:49:16.541 D/dalvikvm( 6783): GC freed 632 objects / 52984 bytes in 87ms
11-27 15:49:22.271 D/dalvikvm(  142): GC freed 1341 objects / 56336 bytes in 
103ms
11-27 15:49:23.141 I/ActivityManager(   97): Displayed activity 
com.xtralogic.android.logcollector/.SendLogActivity: 7219 ms (total 7219 ms)
11-27 15:49:23.201 I/global  ( 6783): Default buffer size used in 
BufferedReader 
constructor. It would be better to be explicit if an 8k-char buffer is required.
11-27 15:49:23.491 D/dalvikvm( 6783): GC freed 1186 objects / 238872 bytes in 
107ms
```

Original comment by `mat.fr...@googlemail.com` on 27 Nov 2009 at 2:55

**Comment by GoogleCodeExporter** [comment]
```
Here;s what I get when running the APK posted above on CM-4.2.5:

E/AndroidRuntime( 1061): Uncaught handler: thread Thread-8 exiting due to 
uncaught
exception
E/AndroidRuntime( 1061): java.lang.RuntimeException: Invalidated. Boom!
E/AndroidRuntime( 1061):    at
tv.studer.smssync.SmsSyncService$2.onInvalidated(SmsSyncService.java:272)
E/AndroidRuntime( 1061):    at
android.database.DataSetObservable.notifyInvalidated(DataSetObservable.java:43)
E/AndroidRuntime( 1061):    at
android.database.AbstractCursor.deactivateInternal(AbstractCursor.java:89)
E/AndroidRuntime( 1061):    at
android.database.AbstractCursor.close(AbstractCursor.java:108)
E/AndroidRuntime( 1061):    at
android.database.BulkCursorToCursorAdaptor.close(BulkCursorToCursorAdaptor.java:
123)
E/AndroidRuntime( 1061):    at 
android.database.CursorWrapper.close(CursorWrapper.java:45)
E/AndroidRuntime( 1061):    at
android.content.ContentResolver$CursorWrapperInner.close(ContentResolver.java:74
6)
E/AndroidRuntime( 1061):    at
tv.studer.smssync.SmsSyncService.backup(SmsSyncService.java:351)
E/AndroidRuntime( 1061):    at
tv.studer.smssync.SmsSyncService.access$2(SmsSyncService.java:240)
E/AndroidRuntime( 1061):    at
tv.studer.smssync.SmsSyncService$1.run(SmsSyncService.java:157)

```

Original comment by `steve.ko...@gmail.com` on 29 Nov 2009 at 9:46

**Comment by GoogleCodeExporter** [comment]
```
@chstuder: I have been experiencing this issue since Cyanogen mod 2.4.5. I just
upgraded to Cyanogen Mod 2.4.6 and the issue still exists.

I uninstalled the SMS Backup app and installed your
SmsSync-CyanogenTesting01-Signed.apk (http://j.mp/6OOye1) from Nov. 27th. It 
failed
again before/after/during the first SMS backup.

Cyanogen says: @kalebdf yeah looking into it. It's an issue on Android 2.0 also.
http://twitter.com/cyanogen/status/6182634074

Thank you for your help and wonderful application. I can't wait to get it up and
running again.
```

Original comment by `kale...@gmail.com` on 29 Nov 2009 at 10:57

**Comment by GoogleCodeExporter** [comment]
```
Issue 56 has been merged into this issue.
```

Original comment by `chstu...@gmail.com` on 30 Nov 2009 at 5:12

**Comment by GoogleCodeExporter** [comment]
```
I have not gotten any confirmation of this issue on stock Android 2.0, changing 
focus
to CyanogenMod only.
```

Original comment by `chstu...@gmail.com` on 30 Nov 2009 at 5:14
- Changed title: **CyanogenMod 4.2.5 / 4.2.6 force-close**

**Comment by GoogleCodeExporter** [comment]
```
I'm on Dwang 1.13 and force closes also. It also happened for me on Cyanogen 
4.2.5
before I had to make the switch to Dwang because of GMail no longer syncing on 
Cyanogen.

```

Original comment by `jeeper...@gmail.com` on 30 Nov 2009 at 7:49

**Comment by GoogleCodeExporter** [comment]
```
Using CM-4.2.6 and SmsSync-CyanogenTesting01-Signed.apk

http://pastebin.com/m6c1936cf 

The juicy bit looks to be:

11-30 21:53:04.738 E/AndroidRuntime(29815): Uncaught handler: thread Thread-8 
exiting
due to uncaught exception
11-30 21:53:04.768 E/AndroidRuntime(29815): java.lang.RuntimeException: 
Invalidated.
Boom!
11-30 21:53:04.768 E/AndroidRuntime(29815): at
tv.studer.smssync.SmsSyncService$2.onInvalidated(SmsSyncService.java:272)
11-30 21:53:04.768 E/AndroidRuntime(29815): at
android.database.DataSetObservable.notifyInvalidated(DataSetObservable.java:43)
11-30 21:53:04.768 E/AndroidRuntime(29815): at
android.database.AbstractCursor.deactivateInternal(AbstractCursor.java:89)
11-30 21:53:04.768 E/AndroidRuntime(29815): at
android.database.AbstractCursor.close(AbstractCursor.java:108)
11-30 21:53:04.768 E/AndroidRuntime(29815): at
android.database.BulkCursorToCursorAdaptor.close(BulkCursorToCursorAdaptor.java:
123)
11-30 21:53:04.768 E/AndroidRuntime(29815): at
android.database.CursorWrapper.close(CursorWrapper.java:45)
11-30 21:53:04.768 E/AndroidRuntime(29815): at
android.content.ContentResolver$CursorWrapperInner.close(ContentResolver.java:74
6)
11-30 21:53:04.768 E/AndroidRuntime(29815): at
tv.studer.smssync.SmsSyncService.backup(SmsSyncService.java:351)
11-30 21:53:04.768 E/AndroidRuntime(29815): at
tv.studer.smssync.SmsSyncService.access$2(SmsSyncService.java:240)
11-30 21:53:04.768 E/AndroidRuntime(29815): at
tv.studer.smssync.SmsSyncService$1.run(SmsSyncService.java:157)
11-30 21:53:04.809 I/Process ( 1370): Sending signal. PID: 29815 SIG: 3

Here's the traces.txt:

----- pid 29815 at 2009-11-30 21:53:04 -----
Cmd line: tv.studer.smssync

DALVIK THREADS:
"main" prio=5 tid=3 TIMED_WAIT
  | group="main" sCount=1 dsCount=0 s=N obj=0x4001e250 self=0xbc20
  | sysTid=29815 nice=0 sched=0/0 cgrp=default handle=-1344001384
  at java.lang.Object.wait(Native Method)
  - waiting on <0x1ac2f0> (a android.os.MessageQueue)
  at java.lang.Object.wait(Object.java:326)
  at android.os.MessageQueue.next(MessageQueue.java:144)
  at android.os.Looper.loop(Looper.java:110)
  at android.app.ActivityThread.main(ActivityThread.java:4203)
  at java.lang.reflect.Method.invokeNative(Native Method)
  at java.lang.reflect.Method.invoke(Method.java:521)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:791)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:549)
  at dalvik.system.NativeStart.main(Native Method)

"Thread-8" prio=5 tid=15 NATIVE
  | group="main" sCount=1 dsCount=0 s=N obj=0x43aa3310 self=0x1e0dc8
  | sysTid=29825 nice=10 sched=0/0 cgrp=bg_non_interactive handle=1706664
  at android.os.BinderProxy.transact(Native Method)
  at
android.app.ActivityManagerProxy.handleApplicationError(ActivityManagerNative.ja
va:2243)
  at com.android.internal.os.RuntimeInit.crash(RuntimeInit.java:302)
  at
com.android.internal.os.RuntimeInit$UncaughtHandler.uncaughtException(RuntimeIni
t.java:75)
  at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:887)
  at java.lang.ThreadGroup.uncaughtException(ThreadGroup.java:884)

"Binder Thread #2" prio=5 tid=13 NATIVE
  | group="main" sCount=1 dsCount=0 s=N obj=0x43a30a20 self=0x15c7f0
  | sysTid=29820 nice=0 sched=0/0 cgrp=default handle=1532848
  at dalvik.system.NativeStart.run(Native Method)

"Binder Thread #1" prio=5 tid=11 NATIVE
  | group="main" sCount=1 dsCount=0 s=N obj=0x43a2db08 self=0x7fff0
  | sysTid=29819 nice=0 sched=0/0 cgrp=default handle=1530448
  at dalvik.system.NativeStart.run(Native Method)

```

Original comment by `steven%m...@gtempaccount.com` on 1 Dec 2009 at 3:00

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the logs!

The patch above changed the app to crash on invalidation of the Cursor that 
queries
the SMS store. It seems strange that it now crashes "regularly" when I close the
Cursor (which invalidates it) but before - with the official 1.1.0 release - it
crashed when iterating over it.

If there's anybody running CyanogenMod who could debug this, that would be 
great!
Unfortunately, I cannot flash my phone and I did not get CyanogenMod running in 
the
emulator.
```

Original comment by `chstu...@gmail.com` on 1 Dec 2009 at 11:12

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
How can I debug it? Not sure how I would be able to do that for you...

Perhaps its something with the xda-messaging.apk instead of the default 
messaging.apk?
```

Original comment by `steven%m...@gtempaccount.com` on 1 Dec 2009 at 1:34

**Comment by GoogleCodeExporter** [comment]
```
My debug, 02/12/2009 
______________________________

12-02 17:23:06.842: INFO/ActivityManager(152): Displayed activity 
tv.studer.smssync/.SmsSync: 2231 ms (total 2231 ms)
12-02 17:23:12.062: INFO/SmsSync(2141): Starting backup...
12-02 17:23:12.102: DEBUG/dalvikvm(29410): GC freed 449 objects / 46944 bytes 
in 
239ms
12-02 17:23:12.262: DEBUG/SmsSync(2141): Total messages to backup: 49
12-02 17:23:12.542: INFO/SmsBackup-Backend(2141): Connecting to imap.gmail.com 
@ IP 
addr imap.gmail.com/72.14.221.111:993
12-02 17:23:13.412: DEBUG/dalvikvm(2141): GC freed 5907 objects / 264856 bytes 
in 
90ms
12-02 17:23:14.262: DEBUG/NetworkLocationProvider(152): onCellLocationChanged 
[30001,28418]
12-02 17:23:14.412: DEBUG/dalvikvm(2141): GC freed 13421 objects / 525360 bytes 
in 
89ms
12-02 17:23:14.852: DEBUG/dalvikvm(2141): GC freed 10442 objects / 437200 bytes 
in 
93ms
12-02 17:23:15.762: DEBUG/dalvikvm(2141): GC freed 9553 objects / 476696 bytes 
in 
96ms
12-02 17:23:16.292: DEBUG/dalvikvm(2141): GC freed 8909 objects / 520536 bytes 
in 
97ms
12-02 17:23:16.872: DEBUG/skia(2006): purging 6K from font cache [1 entries]
12-02 17:23:17.012: DEBUG/dalvikvm(2006): GC freed 2494 objects / 188904 bytes 
in 
127ms
12-02 17:23:17.022: INFO/global(2141): Default buffer size used in 
BufferedOutputStream constructor. It would be better to be explicit if an 8k 
buffer 
is required.
12-02 17:23:18.692: DEBUG/dalvikvm(2141): GC freed 9989 objects / 534240 bytes 
in 
101ms
12-02 17:23:18.912: DEBUG/SmsSync(2141): Sending 1 messages to server.
12-02 17:23:19.292: DEBUG/dalvikvm(2141): GC freed 1408 objects / 92040 bytes 
in 
109ms
12-02 17:23:21.312: DEBUG/SmsSync(2141): Max synced date set to: 1258978608842
12-02 17:23:21.422: ERROR/DatabaseUtils(393): Writing exception to parcel
12-02 17:23:21.422: ERROR/DatabaseUtils(393): java.lang.IllegalStateException: 
attempt to acquire a reference on a close SQLiteClosable
12-02 17:23:21.422: ERROR/DatabaseUtils(393):     at 
android.database.sqlite.SQLiteClosable.acquireReference(SQLiteClosable.java:31)
12-02 17:23:21.422: ERROR/DatabaseUtils(393):     at 
android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:69)
12-02 17:23:21.422: ERROR/DatabaseUtils(393):     at 
android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:288)
12-02 17:23:21.422: ERROR/DatabaseUtils(393):     at 
android.database.sqlite.SQLiteCursor.onMove(SQLiteCursor.java:260)
12-02 17:23:21.422: ERROR/DatabaseUtils(393):     at 
android.database.AbstractCursor.moveToPosition(AbstractCursor.java:188)
12-02 17:23:21.422: ERROR/DatabaseUtils(393):     at 
android.database.CursorToBulkCursorAdaptor.getWindow(CursorToBulkCursorAdaptor.j
ava:1
14)
12-02 17:23:21.422: ERROR/DatabaseUtils(393):     at 
android.database.BulkCursorNative.onTransact(BulkCursorNative.java:66)
12-02 17:23:21.422: ERROR/DatabaseUtils(393):     at 
android.os.Binder.execTransact(Binder.java:287)
12-02 17:23:21.422: ERROR/DatabaseUtils(393):     at 
dalvik.system.NativeStart.run(Native Method)
12-02 17:23:21.452: INFO/ActivityManager(152): Stopping service: 
tv.studer.smssync/.SmsSyncService
12-02 17:23:21.542: DEBUG/SmsSync(2141): Scheduled sync due in 1800 seconds.
12-02 17:23:21.592: WARN/dalvikvm(2141): threadid=15: thread exiting with 
uncaught 
exception (group=0x4001e170)
12-02 17:23:21.602: ERROR/AndroidRuntime(2141): Uncaught handler: thread 
Thread-8 
exiting due to uncaught exception
12-02 17:23:21.662: ERROR/AndroidRuntime(2141): 
java.lang.IllegalStateException: 
attempt to acquire a reference on a close SQLiteClosable
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
android.os.Parcel.readException(Parcel.java:1254)
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:145)
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:111)
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
android.database.BulkCursorProxy.getWindow(BulkCursorNative.java:217)
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
android.database.BulkCursorToCursorAdaptor.onMove(BulkCursorToCursorAdaptor.java
:91)
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
android.database.AbstractCursor.moveToPosition(AbstractCursor.java:188)
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
android.database.AbstractCursor.moveToNext(AbstractCursor.java:256)
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
android.database.CursorWrapper.moveToNext(CursorWrapper.java:191)
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
tv.studer.smssync.CursorToMessage.cursorToMessageArray(CursorToMessage.java:91)
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
tv.studer.smssync.SmsSyncService.backup(SmsSyncService.java:313)
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
tv.studer.smssync.SmsSyncService.access$2(SmsSyncService.java:239)
12-02 17:23:21.662: ERROR/AndroidRuntime(2141):     at 
tv.studer.smssync.SmsSyncService$1.run(SmsSyncService.java:156)
12-02 17:23:21.682: INFO/Process(152): Sending signal. PID: 2141 SIG: 3
12-02 17:23:21.682: INFO/dalvikvm(2141): threadid=7: reacting to signal 3
12-02 17:23:21.812: INFO/DumpStateReceiver(152): Added state dump to 1 crashes
12-02 17:23:21.942: INFO/dalvikvm(2141): Wrote stack trace to 
'/data/anr/traces.txt'
12-02 17:23:29.372: DEBUG/skia(1913): purging 6K from font cache [1 entries]

__________________________________

It seems still a cursor problem...
```

Original comment by `lindiriz...@gmail.com` on 2 Dec 2009 at 4:35

**Comment by GoogleCodeExporter** [comment]
```
The new versions of CM use Dalvik from Eclair.  The problem could be there.
```

Original comment by `steve.ko...@gmail.com` on 2 Dec 2009 at 11:55

**Comment by GoogleCodeExporter** [comment]
```
What about adding a setting to set the maximum number of items per backup to 1? 
Would 
this work?
```

Original comment by `cerebr...@gmail.com` on 3 Dec 2009 at 11:25

**Comment by GoogleCodeExporter** [comment]
```
Well... I set up an eclipse workspace and installed android sdk just to try to 
debug 
it... but I'm not able. The think I could do, however, was to change 
auto-backup timer 
from 30 minutes to 30 seconds and to set a maximum number of items per backup 
to 1. 
This is a good workaround for me.
```

Original comment by `cerebr...@gmail.com` on 3 Dec 2009 at 5:50

**Comment by GoogleCodeExporter** [comment]
```
@Cerebro84 That could be possible. This is always the first app I download 
after I
reset or flash a new rom, THE best app hands down. I like keeping my sms for 
later,
just for references. 
```

Original comment by `saifur...@gmail.com` on 3 Dec 2009 at 6:23

**Comment by GoogleCodeExporter** [comment]
```
Something that solved the issue for me was to change lines 313-314 from 
ConversionResult result = converter.cursorToMessageArray(items,
                        MAX_MSG_PER_REQUEST);

to

ConversionResult result = converter.cursorToMessageArray(items,
                        sItemsToSync);

I'm attaching apk if someone else wants to test... but I'm not responsible for 
any 
damage!! If you don't feel confident you'd better not try it, or edit the 
source and 
compile it yourself
```

Original comment by `cerebr...@gmail.com` on 4 Dec 2009 at 11:53

Attachments:
- [SmsSync_maybeworking.apk](https://storage.googleapis.com/google-code-attachments/android-sms/issue-50/comment-39/SmsSync_maybeworking.apk)

**Comment by GoogleCodeExporter** [comment]
```
tried cerebro84 .apk, the backing up progress are hang after 'logging in' (not 
moving 
from 0), and when i tried to stop, it show 'stopping' for long..
but, after 'canceled' show. suddenly all my items succesfully backed up now. 
wow, 
thanks!
looks like the backup progress are not showing properly.
```

Original comment by `addh...@gmail.com` on 4 Dec 2009 at 1:28

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
No, I forgot to mention, the backup progress bar is lost this way. It goes from 
0 
directly to max when it ends. And stop button losts its funcionality. This is 
just a 
workaround to let backup work while chstuder solves the issue. 
```

Original comment by `cerebr...@gmail.com` on 4 Dec 2009 at 1:34

**Comment by GoogleCodeExporter** [comment]
```
Cerebro84's fix worked for me. Thanks! I see an update to the market version in 
the 
near future. Many happy people! This app is my favorite behind the scenes app 
out 
there. Thanks Chris and Thanks Cerebro for the fix!
```

Original comment by `dmdfreak` on 4 Dec 2009 at 5:23

**Comment by GoogleCodeExporter** [comment]
```
Thanks a lot for your patch cerebro84, it lead me to the real reason of the 
bug, I think!

I was closing the cursor in the finally block within the loop, that is while
iterating over the messages. Older SDK versions seem to have ignored accesses on
closed cursors, but the updated version actually notices and throws an 
exception.

Please have a look at the attached patch and the test signed APK and try it out!

```

Original comment by `chstu...@gmail.com` on 4 Dec 2009 at 8:06

Attachments:
- [cyanogen-testing02.patch](https://storage.googleapis.com/google-code-attachments/android-sms/issue-50/comment-44/cyanogen-testing02.patch)
- [SmsSync-CyanogenTesting02-Signed.apk](https://storage.googleapis.com/google-code-attachments/android-sms/issue-50/comment-44/SmsSync-CyanogenTesting02-Signed.apk)

**Comment by GoogleCodeExporter** [comment]
```
Chris - I can confirm that your most recently posted SmsSync-CyanogenTesting02-
Signed.apk is working on CM 4.2.7.1 . Fast turn around! Nothing short of 
excellence :)
```

Original comment by `salah.ah...@gmail.com` on 4 Dec 2009 at 9:17

**Comment by GoogleCodeExporter** [comment]
```
Chris -
First, MUCH APPRECIATED!  I depend on this app to get all my SMS messages out 
into 
Gmail for later reference.
Second, I am another confirmation that SmsSync-CyanogenTesting02-Signed.apk 
works on 
CyanogenMod 4.2.7.1

On the second manual run of 'Backup Now', the progress bar stopped at 93/100, 
interrupted by an incoming call.  When the call was finished, and SMS Backup 
regained 
focus, the bar was stuck at 93/100.  However, I believe this to be a display 
issue, 
since looking at the logcat, it seemed to send all 100 and finish successfully. 
 I do 
not remember if this happened before.

Also, I did need to uninstall the SMS Backup from the Market to install this 
version 
as 'adb install -r' didn't work.  I am not sure if this is a bigger issue, but 
I 
hadn't had to do that with other programs previously...
```

Original comment by `reukiodo` on 4 Dec 2009 at 10:23

**Comment by GoogleCodeExporter** [comment]
```
Using testing 02 and it's looking good so far, i'll try to report back in a day 
or
two and let you know if anything is force closing...
```

Original comment by `swimmer_...@hotmail.com` on 5 Dec 2009 at 1:38

**Comment by GoogleCodeExporter** [comment]
```
I was playing with filters in Gmail, trying to sort saved texts into separate 
labels 
based on who they came from. When I removed the setting for where to save the 
texts it 
will not log into Gmail. The idea was that if it left the texts unlabeled, the 
filters 
in Gmail would sort them for me. I can't seem to get that part to work, but I 
thought 
you might want to know that leaving that setting blank would keep it from 
backing up 
the messages.

Thanks
```

Original comment by `dmdfreak` on 5 Dec 2009 at 7:27

**Comment by GoogleCodeExporter** [comment]
```
Confirmed working for me too
Happy to have been of help and thank you very much for your great app ;) 
```

Original comment by `cerebr...@gmail.com` on 5 Dec 2009 at 12:07

**Comment by GoogleCodeExporter** [comment]
```
Awesome, thanks for figuring this out.  I've been staring at the source and
scratching my head over it for a few days (I'm cyanogen BTW).  It's one of my
favorite apps :)
```

Original comment by `steve.ko...@gmail.com` on 5 Dec 2009 at 6:31

**Comment by GoogleCodeExporter** [comment]
```
I too confirm the testing02 version works. Thanks.
```

Original comment by `scoobler@gmail.com` on 6 Dec 2009 at 10:07

**Comment by GoogleCodeExporter** [comment]
```
Working here too on: WG-Build 8 which is just a tweaked version of CM 4.2.7.1.
Nice work, great to have backups of these sms's again.
```

Original comment by `AndrewPa...@gmail.com` on 6 Dec 2009 at 1:51

**Comment by GoogleCodeExporter** [comment]
```
The new apk seems to work, although I found a new bug, though it isn't 
seriously.  The 
backup still works.  But now, when the phone goes into sleep mode, the status 
bar 
freezes.  It still uploads all 100 SMSs though.  

After it freezes and finishes uploading, I hit the "Stop Backup" and it starts 
the next 
100 batch.
```

Original comment by `huongk...@gmail.com` on 6 Dec 2009 at 2:13

**Comment by GoogleCodeExporter** [comment]
```
Okay I think I found a fix, sort of.  If the status bar is frozen and there 
isn't any 
data going in an out, just change the orientation of your phone from vertical 
to 
horizontal or vise-versa, and it should refresh the screen.
```

Original comment by `huongk...@gmail.com` on 6 Dec 2009 at 2:26

**Comment by GoogleCodeExporter** [comment]
```
This issue is fixed in revision r 61 and released as version 1.1.1 on the 
Android Market.

Thanks all for the patience and support!

@steve.kondik Good to hear I'm not the only one confused by a simple finally 
clause.
:) Keep up the great work you're doing for the Android community!

@huongkwun I'll file a separate bug for your issue.
```

Original comment by `chstu...@gmail.com` on 6 Dec 2009 at 5:50
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
@huongkwun Bug 57
```

Original comment by `chstu...@gmail.com` on 6 Dec 2009 at 6:03
